### PR TITLE
Misc changes

### PR DESCRIPTION
**Pull request by fingolfin**

### CODE REVIEW

**Comment by markuspf** [comment]
Can you explain why you made parts of the `TypOutputFile` and `TypInputFile` bitfields? For saving space in GAPState?

**Comment by fingolfin** [comment]
Yes, that's the reason -- in fact the idea was to add a few more members in, e.g. the position in the current line doesn't need full 64 / 32 bits etc.
But given that the buffers in these structs make up the vast majority of their size anyway, this might be a pointless "optimization" (and might indeed slow down access?), so I'd be happy to drop this from this PR (or move it to a separate PR, or whatever)

**Comment by fingolfin** [comment]
As always, if people are uncomfortable reviewing this random bunch of commits in a single PR, let me know, and I can try to split it up.

File: src/code.c [hunk 1]
```diff
@@ -119,26 +119,31 @@ static inline void PopLoopNesting( void ) {
   STATE(LoopNesting) = STATE(LoopStack)[--STATE(LoopStackCount)];
 }
 
-static inline void SetupGapname(TypInputFile* i)
```
Comment (ChrisJefferson) [inline src/code.c]: Did you benchmark this? SetupGapname gets called every time we call NewStat, and you are now doing an ELM_LIST on every call.

Comment (ChrisJefferson) [reply src/code.c]: If you want to go down rewriting this code, you could switch back to returning void, and just move the ELM_LIST into a different tiny function, for the places where we want it (that function could call SetupGapname to ensure the values have all been installed).

Comment (fingolfin) [reply src/code.c]: I didn't realize it was also being called by NewStat. I'll change it.

File: src/scanner.c [hunk 2]
```diff
@@ -3045,7 +3045,6 @@ static Int InitKernel (
       StillMoreCookie[i][3] = 'n';  StillMoreCookie[i][4] = 'a';  StillMoreCookie[i][5] = 'm';
       StillMoreCookie[i][6] = 'e';  StillMoreCookie[i][7] = '0'+i;
       StillMoreCookie[i][8] = '\0';
-      InitGlobalBag(&(STATE(InputFiles)[i].gapname), &(StillMoreCookie[i][0]));
```
Comment (ChrisJefferson) [inline src/scanner.c]: If you are getting rid of this, you don't need `StillMoreCookie` anymore.

File: src/code.c [hunk 3]
```diff
@@ -109,24 +109,25 @@ static inline void PopLoopNesting( void ) {
...
-    }
-    else {
-        i->gapnameid = INT_INTOBJ(pos);
-        // Use string from FilenameCache as we know it will not get GCed
-        i->gapname = ELM_LIST( FilenameCache, i->gapnameid );
-    }
+        Obj pos = POS_LIST(FilenameCache, filename, INTOBJ_INT(1));
+        if (pos == Fail) {
+            i->gapnameid = PushPlist(FilenameCache, filename);
+            ;
```
Comment (ChrisJefferson) [inline src/code.c]: Why is there a ; here?

**Comment by fingolfin** [comment]
I moved parts of this PR to two new PRs, #1833 and #1834, and also dropped the bitfield usage in `TypOutputFile` and `TypInputFile`. That should make it easier to review the remaining pieces in this PR.

**Pull request merged**